crates/sui-graphql-rpc/src/data.rs

```diff
@@ -71,33 +67,29 @@ pub(crate) trait DbConnection {
     type Backend: diesel::backend::Backend;
     type Connection: diesel::Connection<Backend = Self::Backend>;
 
-    /// Run a query that fetches a single value. `query` is a thunk that returns a boxed query when
+    /// Run a query that fetches a single value. `query` is a thunk that returns a query when
     /// called.
-    fn result<Q, ST, QS, GB, U>(&mut self, query: Q) -> QueryResult<U>
+    fn result<Q, U>(&mut self, query: impl Fn() -> Q) -> QueryResult<U>
     where
-        Q: Fn() -> Query_<ST, QS, Self::Backend, GB>,
-        Query_<ST, QS, Self::Backend, GB>: LoadQuery<'static, Self::Connection, U>,
-        Query_<ST, QS, Self::Backend, GB>: QueryFragment<Self::Backend>,
-        QS: QuerySource;
+        Q: diesel::query_builder::Query,
+        Q: LoadQuery<'static, Self::Connection, U>,
+        Q: QueryId + QueryFragment<Self::Backend>;
```

Comment:
I see, looks like I was using the wrong syntax when trying this on my own - I was trying to Fn() -> dyn Trait, but this is inaccurate since the closure will always be of the same concrete type for all calls, and I run into object safe issues since rust sees that I'm trying to build a trait object from traits that are not object safe
Tldr til